docs/docs/publishing-component.md

```diff
@@ -0,0 +1,192 @@
+
+### Pre-requisites
+For this section, you will need to have installed [git](https://git-scm.com/downloads) and [node](https://nodejs.org/en/download/)
+
+For this guide we will create a `PageTitle` react component and would lay out the necessary steps for publishing it to [npm](https://www.npmjs.com/)
+
+### Instructions
+* Create a project directory.
+* Open terminal in the project directory. 
+* React components are published as npm packages, so we need a `package.json` file. To create one, we will initialize our project directory with a `package.json`
+```bash
+npm init
+```
+This would walk us through creating our `package.json` file.
+
+* We will install [react](https://facebook.github.io/react/). We would save it as a dependency to our package.
+```bash
+npm i -S react
+```
+
+* Next, we need some development tools to assist us in development and for publishing a release. We will use [Babel](https://babeljs.io/) for transpiling our ES6 code to ES5. Also, for coverage we would use [Jest](https://facebook.github.io/jest/) and [Enzyme](http://airbnb.io/enzyme/).
+```bash
+npm i -D babel-cli babel-preset-es2015 babel-preset-react babel-jest enzyme jest react-addons-test-utils
+```
+
+* Next we will configure babel presets for the transforms that we want to include for our transpilation. There are multiple ways to achieve this. We will create a new file at root level named as `.babelrc`
+```
+{
+  "presets": ["es2015", "react"]
+}
+```
+
+*For alternative ways to configure babel preset options* [babelrc](https://babeljs.io/docs/usage/babelrc/)
+
+* In our package json, we have an option as `main` that basically is set against a file in our source. So when someone installs our package into their own application and imports our package for usage, for example:
+
+``` javascript
+import PageTitle from 'react-page-title';
+```
+The path for file that we have specified would be the file it would import contents from. Currently in our package, it points to `src/index.js`.
+
+* We will now update our `npm scripts` in our `package.json` file
+
+``` javascript
+"scripts": {
+    "build": "babel --out-dir dist --ignore *.test.js src",
+    "test": "jest"
+}
+```
+
+Now, if we run `npm run build` as we have instructed babel, it would create a `dist` directory for us and would transpile every file inside our `src` directory. Also, as we do not want our test files to be traspiled we have added an `--ignore` option to our build script for ignoring all files that ends with `.test.js`.
+Moreover, we have added a script for running our tests. 
+ 
+* Since we do not want others to import our ES6 version of component when they import our package in their applications, we would need to update our `main` option in our `package.json` to point to the transpiled version of our component. Our scripts and name become
+ ``` javascript
+  "main": "dist/index.js", // Now points to the transpiled version
+   "scripts": {
```

Comment:
Closed this for https://github.com/facebook/react/pull/8439 @brigand Indentation fixed!